petab/lint.py

```diff
@@ -832,6 +832,9 @@ def is_valid_identifier(x: str) -> bool:
     Returns:
         ``True`` if valid, ``False`` otherwise
     """
+    # Missing IDs are float('nan')
+    if isinstance(x, float) and np.isnan(x):
```

Comment:
```suggestion
    if pd.isan(x):
```
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.isna.html#pandas.isna

Should cover a couple of additional cases.
(from numbers import Number)